kolibri/plugins/document_epub_render/assets/src/views/EpubRendererIndex.vue

```diff
@@ -418,35 +382,75 @@
       global.ePub = Epub;
       this.book = new Epub(this.epubURL);
 
-      const { savedTheme = THEMES.WHITE, savedFontSize } =
+      const { theme = this.theme, fontSize = this.fontSize } =
         Lockr.get(EPUB_RENDERER_SETTINGS_KEY) || {};
-      this.theme = savedTheme;
-      this.fontSize = savedFontSize;
+      this.theme = theme;
+      this.fontSize = fontSize;
     },
     mounted() {
       Promise.all([this.cssPromise, this.book.ready]).then(() => {
         if (this.book.navigation) {
           this.toc = this.book.navigation.toc;
         }
 
-        const width = this.calculateRenditionWidth(this.elementWidth);
-        const height = this.calculateRenditionHeight(this.elementHeight);
         this.rendition = this.book.renderTo(this.$refs.epubjsContainer, {
-          defaultManager,
           view: iFrameView,
-          width,
-          height,
+          resizeOnOrientationChange: false,
           spread: 'auto',
           minSpreadWidth: 600,
         });
 
-        if (this.savedLocation) {
-          this.rendition
-            .display(this.savedLocation)
-            .then(() => this.handleReadyRendition(width, height));
-        } else {
-          this.rendition.display().then(() => this.handleReadyRendition(width, height));
-        }
+        this.rendition
+          .display(this.savedLocation || undefined)
+          .then(() => {
+            const hasTables = this.rendition.getContents().reduce((hasTable, contents) => {
+              return hasTable || contents.document.getElementsByTagName('table').length > 0;
+            }, false);
+
+            // Tables have issues rendering, so switch from paginated flow to scrolled
+            if (hasTables) {
+              this.scrolled = true;
+              this.rendition.flow('scrolled');
+              this.rendition.clear();
+              // Re-renders and targets this.savedLocation (if set)
+              return this.rendition.display(this.savedLocation || undefined);
+            }
```

Comment:
Would you mind posting a quick comparison of epubs rendered as scrolled vs paginated? I'm curious how big a difference it "feels like".

If there's a chance that having two different interaction modes is confusing, perhaps it would be preferable for consistency to display _all_ books as scrolled? (cc @jtamiace @khangmach)

Reply:
The paginated experience feels more like a book because the content is usually rendered with two columns, with the exception of smaller viewports where it will transition to a single column. I feel that the paginated mode is superior for reading books and pure text documents since the readability is better, specifically because of the column layout.

The scrolled mode is definitely superior for documents with tables or inflexible flow typed content that doesn't morph to different layouts. We may find there are other situations where the scrolled mode is superior as well, but when it comes to books, I think it's a poorer experience. To render a book in scrolled mode, it would be preferable to limit the content width which wastes screen real-estate and essentially makes fullscreen mode mostly unnecessary.

I don't think it's confusing to have different modes. Notion essentially does this with tables inside pages, but contextualizes the scrolled vs. paginated/column mode by each block. Mixing the two in the same document I find actually less ideal, so this approach is the best of all options IMO.

Reply:
sounds good!